CRM/Upgrade/Incremental/php/FiveTwelve.php

```diff
@@ -84,4 +96,66 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
   //   return TRUE;
   // }
 
+  /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_5_12_alpha1($rev) {
+    $this->addTask(ts('Upgrade DB to %1: SQL', array(1 => $rev)), 'runSql', $rev);
+    $this->addTask('Migrate to utf8mb4', 'migrateUtf8mb4');
+  }
+
+  /**
+   * Performs the utf8mb4 migration.
+   */
+  public static function migrateUtf8mb4(CRM_Queue_TaskContext $ctx) {
+    $dao = new CRM_Core_DAO();
+    $database = $dao->_database;
+    CRM_Core_DAO::executeQuery("ALTER DATABASE $database CHARACTER SET = utf8mb4 COLLATE = utf8mb4_unicode_ci");
+    $dao = CRM_Core_DAO::executeQuery("SHOW TABLE STATUS WHERE (Engine = 'InnoDB' OR Engine = 'ARCHIVE') AND (Name LIKE 'civicrm\_%' OR Name LIKE 'log\_civicrm\_%')");
+    while ($dao->fetch()) {
+      $tables[$dao->Name] = $dao->Engine;
+    }
+    foreach ($tables as $table => $engine) {
+      if ($engine === 'InnoDB') {
+        // Disable i18n rewrite.
+        CRM_Core_DAO::executeQuery("ALTER TABLE $table ROW_FORMAT = Dynamic", array(), TRUE, NULL, FALSE, FALSE);
+      }
+      // Disable i18n rewrite.
+      CRM_Core_DAO::executeQuery("ALTER TABLE $table CHARACTER SET = utf8mb4 COLLATE = utf8mb4_unicode_ci", array(), TRUE, NULL, FALSE, FALSE);
+      $dao = CRM_Core_DAO::executeQuery("SHOW FULL FIELDS FROM $table", array(), TRUE, NULL, FALSE, FALSE);
+      while ($dao->fetch()) {
+        $collation = 'utf8mb4_unicode_ci';
+        $params = array(1 => array($dao->Comment, 'String'));
+        if (!$dao->Collation || $dao->Collation === $collation) {
+          continue;
+        }
+        if (strpos($dao->Collation, 'utf8') !== 0) {
+          continue;
+        }
+        if (strpos($dao->Collation, '_bin') !== FALSE) {
+          $collation = 'utf8mb4_bin';
+        }
+        if ($dao->Null === 'YES') {
```

Comment:
@mfb why are you setting the variable $null here when it doesn't appear to be used again?